tests/tt_metal/tt_metal/perf_microbenchmark/1_compute_mm/test_compute_mm.cpp

```diff
@@ -559,6 +851,183 @@ tuple<uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t, uint32_t> get_
+    uint32_t in0_block_num_tiles = out_subblock_h * Kt * in0_num_subblocks;
+    uint32_t in0_subblock_num_tiles = out_subblock_h * Kt;
+
+    uint32_t in1_num_subblocks = (Nt / out_subblock_w);
+    uint32_t in1_block_num_tiles = out_subblock_w * Kt * in1_num_subblocks;
+    uint32_t in1_per_core_w = out_subblock_w * in1_num_subblocks;
+
+    uint32_t out_subblock_num_tiles = out_subblock_h * out_subblock_w;
+
+    vector<uint32_t> compute_kernel_args = {
+        Kt,             // in0_block_w
+        in0_num_subblocks,       // in0_num_subblocks
+        in0_block_num_tiles,     // in0_block_num_tiles
+        in0_subblock_num_tiles,  // in0_subblock_num_tiles
+
+        in1_num_subblocks,    // in1_num_subblocks
+        in1_block_num_tiles,  // in1_block_num_tiles
+        in1_per_core_w,       // in1_per_core_w
+
+        num_blocks,  // num_blocks
+
+        out_subblock_h,          // out_subblock_h
+        out_subblock_w,          // out_subblock_w
+        out_subblock_num_tiles,  // out_subblock_num_tiles
+        1,                        // batch
+        Mt*Nt
+    };
+
+    vector<uint32_t> reader_kernel_args = {
+        Mt*Kt,
+        num_blocks,
+    };
+    vector<uint32_t> writer_kernel_args = {
+        Nt*Kt,
+        num_blocks,
+    };
+
+    log_debug("in0_cb_addr: {}", (*in0_cb_addr).address());
+    log_debug("in1_cb_addr: {}", (*in1_cb_addr).address());
+    log_debug("out_cb_addr: {}", (*out_cb_addr).address());
+    log_debug("cb_data_format: {}", cb_data_format);
+    log_debug("single_tile_size: {}", single_tile_size);
+    if (matmul_block) {
+        log_debug("matmul_block");
+    } else {
+        log_debug("matmul_tiles");
+    }
+    if (packer_l1) {
+        log_debug("packer_l1");
+    } else {
+        log_debug("no packer_l1");
+    }
+
+    CoreRange all_cores(
+        {(std::size_t)0, (std::size_t)0}, {(std::size_t)core_range.x - 1, (std::size_t)core_range.y - 1});
+
+    // Create circular buffers
+    uint32_t src0_cb_index = 0;
+    tt_metal::CircularBufferConfig cb_src0_config =
+        tt_metal::CircularBufferConfig(in0_CB_tiles * single_tile_size, {{src0_cb_index, cb_data_format}})
+            .set_page_size(src0_cb_index, single_tile_size)
+            .set_globally_allocated_address(*in0_cb_addr);
+    auto cb_src0 = tt_metal::CreateCircularBuffer(program, all_cores, cb_src0_config);
+
+    uint32_t src1_cb_index = 1;
+    tt_metal::CircularBufferConfig cb_src1_config =
+        tt_metal::CircularBufferConfig(in1_CB_tiles * single_tile_size, {{src1_cb_index, cb_data_format}})
+            .set_page_size(src1_cb_index, single_tile_size)
+            .set_globally_allocated_address(*in1_cb_addr);
+    auto cb_src1 = tt_metal::CreateCircularBuffer(program, all_cores, cb_src1_config);
+
+    uint32_t out_cb_index = 16;  // output operands start at index 16
+    uint32_t interm0_cb_index = 24;
+
+    if (packer_l1 and cb_data_format == tt::DataFormat::Bfp8_b) {
```

Comment:
When packer_l1 is true and dtype is Bfp8_b, the dtype of intermed cb is Float16_b not Bfp8_b, so adding some logic for this in get_in0_block_w function is required.

Reply:
get_in0_block_w is not required, in0_block_w is equal to Kt in the sharded matmul, and we use num_blocks to repeatedly send n blocks to the kernel.